python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -332,9 +332,13 @@ def run(self, science, matchedTemplate, difference,
         if idFactory is None:
             idFactory = lsst.meas.base.IdGenerator().make_table_id_factory()
 
-        # Ensure that we start with an empty detection mask.
+        # Ensure that we start with an empty detection and deblended mask.
         mask = difference.mask
-        mask &= ~(mask.getPlaneBitMask("DETECTED") | mask.getPlaneBitMask("DETECTED_NEGATIVE"))
+        clearMaskPlanes = ["DETECTED", "DETECTED_NEGATIVE", "NOT_DEBLENDED"]
+        for mp in clearMaskPlanes:
+            if mp not in mask.getMaskPlaneDict():
+                mask.addMaskPlane(mp)
```

Comment:
I understand why you're doing this, but it makes me a bit uncomfortable. We should be getting a coherent mask plane dict from earlier in the pipeline, if we're not that's a problem.

At the very least a comment above these lines saying why you're adding them would be good.